OpenSim/Common/ArrayPtrs.h

```diff
@@ -1024,6 +1064,121 @@ int searchBinary(const T &aObject,bool aFindFirst=false,
 //=============================================================================
 };  // END of class ArrayPtrs
 
+
+//=============================================================================
+// Custom Iterator
+//=============================================================================
+//_____________________________________________________________________________
+
+/**
+* Inner class that describes a const_iterator and 'regular' iterator at the same time,
```

Comment:
If I make it inner, there are some template initializations that produce errors:

https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/Object.h#L54
https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/FunctionSet.cpp#L41